fe/fe-core/src/main/java/com/starrocks/sql/optimizer/cost/CostModel.java

```diff
@@ -129,6 +130,15 @@ public CostEstimate visitOperator(Operator node, ExpressionContext context) {
         public CostEstimate visitPhysicalOlapScan(PhysicalOlapScanOperator node, ExpressionContext context) {
             Statistics statistics = context.getStatistics();
             Preconditions.checkNotNull(statistics);
+            if (node.getTable().isMaterializedView()) {
+                ColumnRefSet usedColumns = statistics.getUsedColumns();
+                Projection projection = node.getProjection();
+                if (projection != null) {
+                    // remove projection keys
+                    usedColumns.except(projection.getColumnRefMap().keySet());
+                }
+                return CostEstimate.of(statistics.getOutputSize(usedColumns), 0, 0);
```

Comment:
used column contains predicate columns? if not, why not compute this cost?

Reply:
yes, used column contains predicate columns.